main.go

```diff
@@ -462,3 +487,207 @@ func stopStalePipelines (client *gitlab.Client, vars []*gitlab.PipelineVariable)
+			if err != nil {
+				return fmt.Errorf("syncIfOSHasEnterpriseRepo: Failed to create the PR branch on the Enterprise repo due to error: %v", err)
+			}
+
+			pr, err := createPullRequestFromEnterpriseBranch(createPRArgs{
+				repo:        repo.GetName() + "-enterprise",
+				prBranch:    PRBranchName,
+				baseBranch:  branchRef,
+				accessToken: conf.osToEntGithubToken,
+				message:     fmt.Sprintf("[Bot] %s", pr.GetTitle()),
+				messageBody: pr.GetBody(),
+				dryRun:      true,
+			})
+			if err != nil {
+				return fmt.Errorf("Failed to create a PR with error: %v", err)
+			}
+
+			log.Infof("syncIfOSHasEnterpriseRepo: Created PR: %v", pr)
+
+		}
+
+	}
+
+	return nil
+}
+
+// createPRBranchOnEnteprise creates a new branch in the Enterprise repository
+// starting at the branch in which to sync, with the name 'PRBranchName'
+// and merges this with the OS equivalent of 'branchName'.
+func createPRBranchOnEnterprise(repo, branchName, PRBranchName string) error {
+
+	tmpdir, err := ioutil.TempDir("", repo)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpdir)
+
+	gitcmd := exec.Command("git", "init", ".")
+	gitcmd.Dir = tmpdir
+	out, err := gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	gitcmd = exec.Command("git", "remote", "add", "opensource", "git@github.com:mendersoftware/"+repo+".git")
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	gitcmd = exec.Command("git", "remote", "add", "enterprise", "git@github.com:mendersoftware/"+repo+"-enterprise"+".git")
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Fetch the branch which we are going to sync
+	gitcmd = exec.Command("git", "fetch", "opensource", branchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Fetch the Enterprise branch in which to merge into, and create the PR branch
+	gitcmd = exec.Command("git", "fetch", "enterprise", branchName+":"+PRBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Checkout the enterprise PR branch
+	gitcmd = exec.Command("git", "checkout", PRBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Merge the branches
+	mergeMessage := fmt.Sprintf("Merge branch '%s' into '%s'", "opensource/"+branchName, "enterprise/"+PRBranchName)
+	gitcmd = exec.Command("git", "merge", "-m", mergeMessage, "opensource/"+branchName)
```

Comment:
Mmm. Do we know what will happen on merge conflicts? I haven't tested, but I think this command will return error...

What I think we should do? I think we should force a merge even with conflicts, so that a PR with conflicts is launched to GitHub. That will not work (i.e. will not pass any build), but we as developers will see it and fix. If the bot does not launch a PR, the humans will ignore the problem until is too late :)

Reply:
I guess this relates to the `MaintainersCanModify` above. So good point I guess.

Reply:
The way it is now, it will simply fail.

Reply:
Spoke to Kristian about it, and he claims the merge is unnecessary, as github will handle it. So maybe we should drop the merge fully?

Reply:
I don't fully understand, but I take the Git Guru word here.

If we can avoid the merge the git history will look much cleaner, and the possibility for the merge conflict is out. Good.

Reply:
When someone says he is waiting for the next release of git, for some feature he is missing, you trust him :p